app/controllers/pieces_controller.rb

```diff
@@ -7,17 +10,34 @@ def show
 
   def update
     @selected_piece = ChessPiece.find(params[:id])
+    @game = @selected_piece.game
 
-    if @selected_piece && @selected_piece.valid_move?(Coordinates.new(move_to_x_parameter, move_to_y_parameter))
+    if moving_validly?
       @selected_piece.move_to!(Coordinates.new(move_to_x_parameter, move_to_y_parameter))
-      redirect_to piece_path(@selected_piece)
+      redirect_to game_path(@game)
     else
-      render 'Invalid', status: :unprocessable_entity
+      render text: 'Forbidden', status: :unauthorized
     end
   end
 
   private
 
+  def authorize_player
+    render text: 'Forbidden', status: :unauthorized unless current_game.players.include?(current_user)
+  end
+
+  def current_game
+    @game ||= ChessPiece.find(params[:id]).game
+  end
+
+  def players
+    [current_game.black_player, current_game.white_player]
+  end
+
+  def moving_validly?
+    @selected_piece && @selected_piece.valid_move?(Coordinates.new(move_to_x_parameter, move_to_y_parameter))
```

Comment:
I don't think the nil check is necessary here. ChessPiece.find throws an error if it doesn't find a piece.